controllers/ingress_controller.go

```diff
@@ -221,6 +221,10 @@ func (r *IngressReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	return ctrl.Result{}, nil
 }
 
+func (r *IngressReconciler) buildAPIName(nameSpace, name, hostName, path string) string {
+	return strings.TrimRight(strings.ReplaceAll(fmt.Sprintf("%s-%s-%s-%s", nameSpace, name, hostName, path), "/", ""), "-")
```

Comment:
Looking at this I am thinking we can play it a bit safer. Because path docs says

> 	// Path is matched against the path of an incoming request. Currently it can
	// contain characters disallowed from the conventional "path" part of a URL
	// as defined by RFC 3986. Paths must begin with a '/'. When unspecified,
	// all paths from incoming requests are matched.
	// +optional

A simple replacement of `/` with `""` won't do and much and will complicate things

```suggestion
	return fmt.Sprintf("%s-%s-%s", nameSpace, name, encodeNS(hostName+path))
```

This will be just `{namespace}-{name}-{base64 encoded combo of host+path}`

we can revisit this any other time if requirements changes.

Reply:
```
a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')"
```

For the purposes of this PR - I will leave as-is, but we will need to fix it further down the line for sure. e.g. we don't currently support regexp in the listen path for the operator and im not sure we will need to.

Reply:
> we don't currently support regexp in the listen path for the operator and im not sure we will need to
Do we have a validation for this? And `listen_path` is bread and butter of the gateway apidef. This is a very bold statement.

Reply:
Ingress doesn't allow regexp in the listen path. So we wouldn't need to do it for ApiDefinition either.